ocaml/testsuite/tests/typing-layouts/error_message_attr.ml

```diff
@@ -0,0 +1,118 @@
+end
+
+[%%expect{|
+Line 3, characters 61-63:
+3 |   val f : (('a : value)[@error_message "Custom message"]) -> 'a t
+                                                                 ^^
+Error: This type ('a : value) should be an instance of type ('b : float64)
+       The layout of 'a is value, because
+         the [@error_message] attribute: Custom message.
+       But the layout of 'a must overlap with float64, because
+         of the definition of t at line 2, characters 2-28.
+|}]
+
+
+(* Ltyp_alias { aliased_type; name; layout } case *)
+
+(* First call to [layout_of_annotation_or_attr] in [transl_type_alias] *)
+module type a = sig
+  type t : float64
+  val f : 'a -> t -> (t as ('a : value)[@error_message "Custom message"])
+end
+[%%expect{|
+Line 3, characters 33-38:
+3 |   val f : 'a -> t -> (t as ('a : value)[@error_message "Custom message"])
+                                     ^^^^^
+Error: Bad layout annotation:
+         The layout of t is float64, because
+           of the definition of t at line 2, characters 2-18.
+         But the layout of t must be a sublayout of value, because
+           the [@error_message] attribute: Custom message.
+|}]
+
+(* Second call to [layout_of_annotation_or_attr] in the Not_found case
+   of [transl_type_alias] *)
+module type a = sig
+  type t : float64
+  val f : t -> (t as ('a : value)[@error_message "Custom message"])
+end
+[%%expect{|
+Line 3, characters 16-33:
+3 |   val f : t -> (t as ('a : value)[@error_message "Custom message"])
+                    ^^^^^^^^^^^^^^^^^
+Error: This alias is bound to type t but is used as an instance of type
+         ('a : value)
+       The layout of t is float64, because
+         of the definition of t at line 2, characters 2-18.
+       But the layout of t must be a sublayout of value, because
+         the [@error_message] attribute: Custom message.
+|}]
+
+(* Third call to [layout_of_annotation_or_attr] in the None case
+   of [transl_type_alias] *)
+module type a = sig
+  type t : float64
+  val f : t -> (t as (_ : value)[@error_message "Custom message"])
+end
+[%%expect{|
+Line 3, characters 32-65:
+3 |   val f : t -> (t as (_ : value)[@error_message "Custom message"])
+                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Warning 47 [attribute-payload]: illegal payload for attribute 'error_message'.
+More than one error_message attribute present. All of them will be ignored.
+Line 3, characters 26-31:
+3 |   val f : t -> (t as (_ : value)[@error_message "Custom message"])
+                              ^^^^^
+Error: Bad layout annotation:
+         The layout of t/2 is float64, because
+           of the definition of t at line 2, characters 2-18.
+         But the layout of t/2 must be a sublayout of value, because
+           of the annotation on the wildcard _ at line 3, characters 26-31.
+|}]
+(* CR layouts: Bug in Jane_syntax is duplicating the attrs here *)
```

Comment:
Crumbs. But actually I think the new approach to extract the attributes will fix this. Maybe. Try that and resolve this comment if it does actually fix the problem.

Reply:
The new approach does fix this. Something to note is that since the duplicated attributes are not registered, you also don't get the `warning 53` misplaced-attribute error from them.

Reply:
Right. That's exactly what I was hoping for. Sometimes, two wrongs _do_ make a right. :)